src/day2/scan_passwds.rb

```diff
@@ -0,0 +1,95 @@
+#! /usr/bin/env -S ruby -w
+
+#: AoC Day 2 - Count valid passwords. (See `problems.txt` for the exact
+#: problems definitions.)
+
+require 'optparse'
+require 'ostruct'
+
+module ScanPasswds
+
+  #: Return true if given passwd matches original shopkeeper's policies
+  #: constraints, false otherwise.
+  def ScanPasswds.match_constraints_orig?(min, max, char, passwd)
+    count = passwd.count char
+    min <= count && count <= max
+  end
+
+  #: Return true if given passwd matches Toboggan Corporate Policies constraints,
+  #: false otherwise.
+  def ScanPasswds.match_constraints_tcp?(fst_idx, sec_idx, char, passwd)
+    (passwd[fst_idx - 1] == char) ^ (passwd[sec_idx - 1] == char)
+  end
+
+  #: Return true if given passwd matches Toboggan Corporate Policies constraints,
+  #: false otherwise.
+  def ScanPasswds.passwd_valid?(line, match_method_sym)
+    match_data = line.strip.match(/^(\d+)-(\d+)\s+(\w):\s+(.+)$/)
+    raise "Line does NOT match format: [#{line.strip}]" unless match_data
+    min_s, max_s, char, passwd = match_data.captures
+    min, max = min_s.to_i, max_s.to_i
+    # print "min=#{min}, max=#{max}, char=#{char}, passwd=#{passwd}\n"
+    match_method = ScanPasswds.method(match_method_sym)
```

Comment:
Clever! But we don't like clever code :) An `if` would've been sufficient.

Reply:
Indeed. Fixed in df1b427. Also fixed command line options propagation in `ExpenseReportFix` the same way.